app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-enhanced-ecommerce-tracker.js

```diff
@@ -0,0 +1,133 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  'use strict'
+
+  function Ga4EnhancedEcommerceTracker (module) {
+    this.module = module
+  }
+
+  Ga4EnhancedEcommerceTracker.prototype.init = function (isNewPage = true) {
+    var consentCookie = window.GOVUK.getConsentCookie()
+    this.isNewPage = isNewPage
+
+    if (consentCookie && consentCookie.settings) {
+      this.startModule()
+    } else {
+      this.startModule = this.startModule.bind(this)
+      window.addEventListener('cookie-consent', this.startModule)
+    }
+  }
+
+  Ga4EnhancedEcommerceTracker.prototype.startModule = function () {
+    this.PIIRemover = new window.GOVUK.analyticsGA4.PIIRemover()
+    this.DEFAULT_LIST_TITLE = 'Site search results'
+
+    window.dataLayer = window.dataLayer || []
```

Comment:
I think we should try to avoid initialising the dataLayer inside our code. Instead, I think we should be checking to see whether it exists as an early step, and backing out if it doesn't (on the assumption that GTM has failed to initialise for some reason, perhaps because cookie consent hasn't been given).

Reply:
Agreed, I've changed this to just be `if(window.dataLayer){ ... }`